--- a/configure.ac
+++ b/configure.ac
@@ -5,7 +5,7 @@ AC_CONFIG_SRCDIR([gphoto2/main.c])
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIR([auto-m4])
 AC_CONFIG_AUX_DIR([auto-aux])
-AM_INIT_AUTOMAKE([-Wall -Werror gnu 1.9 dist-bzip2 check-news])
+AM_INIT_AUTOMAKE([-Wall gnu 1.9 dist-bzip2 check-news])
 
 if test ! -d "$srcdir/gphoto-m4"; then
 AC_MSG_ERROR([
@@ -32,7 +32,7 @@ AC_PROG_CC
 AC_PROG_CPP
 AM_PROG_CC_C_O
 m4_pattern_allow([AM_PROG_AR], [AM_PROG_AR])
-AM_PROG_LIBTOOL
+LT_INIT
 AC_PROG_INSTALL
 AC_SYS_LARGEFILE
 GP_PKG_CONFIG
@@ -60,7 +60,6 @@ GP_GETTEXT_HACK([],[Lutz Müller and oth
 ALL_LINGUAS="az cs da de en_GB es eu fi fr hu id is it ja nl pa pl pt_BR ro ru rw sk sr sv uk vi zh_CN zh_TW"
 AM_GNU_GETTEXT_VERSION([0.14.1])
 AM_GNU_GETTEXT([external])
-AM_PO_SUBDIRS()
 AM_ICONV()
 GP_GETTEXT_FLAGS()
 
@@ -354,20 +353,13 @@ AC_SUBST([AM_LDFLAGS])
 # Create output files
 # ---------------------------------------------------------------------------
 AC_CONFIG_FILES([
-po/Makefile.in 
 Makefile
 gphoto2/Makefile
 gphoto-m4/Makefile
-doc/Makefile
 contrib/Makefile
 packaging/Makefile
 packaging/rpm/Makefile
 packaging/rpm/package.spec
-tests/data/Makefile
-tests/staging/subdir1/Makefile
-tests/staging/subdir2/Makefile
-tests/staging/Makefile
-tests/Makefile
 ])
 AC_OUTPUT()dnl
 
--- a/gphoto2/Makefile.am
+++ b/gphoto2/Makefile.am
@@ -9,8 +9,6 @@ EXTRA_DIST = \
 
 bin_PROGRAMS = gphoto2
 
-doc_DATA = test-hook.sh
-
 if HAVE_CDK
 CDK_FILES = gphoto2-cmd-config.c gphoto2-cmd-config.h
 else
@@ -50,10 +48,4 @@ gphoto2_LDADD =								\
 	$(LIBGPHOTO2_LIBS) $(CDK_LIBS) $(AA_LIBS) $(JPEG_LIBS) 		\
 	$(PTHREAD_LIBS) $(LIBEXIF_LIBS) $(INTLLIBS) $(RL_LIBS) $(POPT_LIBS)
 
-TESTS = spawntest
-
-TESTS_ENVIRONMENT = GPHOTO_HOOK=$(srcdir)/test-hook.sh
-
-check_PROGRAMS = spawntest
 
-spawntest_SOURCES = spawntest.c spawnve.c spawnve.h
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = gphoto-m4 contrib doc gphoto2 packaging po tests
+SUBDIRS = gphoto-m4 contrib gphoto2 packaging
 
 ACLOCAL_AMFLAGS = -I auto-m4 -I gphoto-m4
 EXTRA_DIST = README.md
