--- a/hexdump.c
+++ b/hexdump.c
@@ -91,9 +91,9 @@ void hexdump (const void *mem, int bytes
         bytes_left >= format->bytes_per_line;
         bytes_left -= format->bytes_per_line)
     { /* print start address for current line */
-      fprintf(outfile, format->indent);
+      fprintf(outfile, "%s", format->indent);
       addrprint(outfile, addrprint_start + index, format->addrprint_width);
-      fprintf(outfile, format->sep1);
+      fprintf(outfile, "%s", format->sep1);
 
       /* display the bytes in hex */
       for (i = 0; ; )
@@ -102,11 +102,11 @@ void hexdump (const void *mem, int bytes
          if (++i >= format->bytes_per_line)
             break;
 
-         fprintf(outfile, format->sep2);
+         fprintf(outfile, "%s", format->sep2);
        }
 
       index -= format->bytes_per_line;
-      fprintf(outfile, format->sep3);
+      fprintf(outfile, "%s", format->sep3);
 
       /* display the bytes as characters */
       for (i = 0; i < format->bytes_per_line; i++)
@@ -120,14 +120,14 @@ void hexdump (const void *mem, int bytes
       return;
 
    /* print start address for last line */
-   fprintf(outfile, format->indent);
+   fprintf(outfile, "%s", format->indent);
    addrprint(outfile, addrprint_start + index, format->addrprint_width);
-   fprintf(outfile, format->sep1);
+   fprintf(outfile, "%s", format->sep1);
 
    /* display bytes for last line in hex */
    for (i = 0; i < bytes_left; i++)
     { hexprint(outfile, p[index++]);
-      fprintf(outfile, format->sep2);
+      fprintf(outfile, "%s", format->sep2);
     }
 
    index -= bytes_left;
@@ -139,10 +139,10 @@ void hexdump (const void *mem, int bytes
       if (++i >= format->bytes_per_line)
          break;
 
-      fprintf(outfile, format->sep2);
+      fprintf(outfile, "%s", format->sep2);
     }
 
-   fprintf(outfile, format->sep3);
+   fprintf(outfile, "%s", format->sep3);
 
    /* display bytes for last line as characters */
    for (i = 0; i < bytes_left; i++)
