--- a/Makefile
+++ b/Makefile
@@ -154,7 +154,7 @@ endif
 
 CPPFLAGS=		$(CPPFLAGS.l2tptest) $(CPPFLAGS-y)
 CFLAGS=			-I. -Iusl -Icli -isystem include \
-				-MMD -Wall -Werror -Wno-strict-aliasing \
+				-MMD -Wall -Wno-strict-aliasing \
 				$(CPPFLAGS) $(CPPFLAGS.dmalloc) \
 				-DSYS_LIBDIR=$(SYS_LIBDIR)
 LDFLAGS.l2tpd=		-Wl,-E -L. -Lusl -lusl -ldl $(LIBS.dmalloc) -lc
--- a/cli/Makefile
+++ b/cli/Makefile
@@ -7,7 +7,7 @@ CLI_SRCS_TEST.o=	$(CLI_SRCS_TEST.c:%.c=%
 
 LDFLAGS.cli_test=	-L.. -L. $(READLINE_LDFLAGS) -lcli -lusl -lreadline -lcurses -lc
 
-CFLAGS=			$(CFLAGS.optimize) -MMD -Wall -Werror -I.. $(READLINE_CFLAGS)
+CFLAGS=			$(CFLAGS.optimize) -MMD -Wall -I.. $(READLINE_CFLAGS)
 
 .PHONY:			all test clean
 
--- a/usl/Makefile
+++ b/usl/Makefile
@@ -14,7 +14,7 @@ LIBS.dmalloc=		-ldmalloc
 endif
 
 CPPFLAGS+=		$(CFLAGS.optimize) -g
-CFLAGS=			-I. -MMD -Wall -Werror -Wno-strict-aliasing $(CPPFLAGS) $(CPPFLAGS.dmalloc)
+CFLAGS=			-I. -MMD -Wall -Wno-strict-aliasing $(CPPFLAGS) $(CPPFLAGS.dmalloc)
 
 .PHONY:			all test clean
 
