--- a/config.h.in
+++ b/config.h.in
@@ -10,6 +10,9 @@
 // Define to the address where bug reports for this package should be sent.
 #define USTL_BUGREPORT	"@PKG_BUGREPORT@"
 
+/// Define to 1 if you want to build without libstdc++
+#undef WITHOUT_LIBSTDCPP
+
 /// Define to 1 if you want stream operations to throw exceptions on
 /// insufficient data or insufficient space. All these errors should
 /// be preventable in output code; the input code should verify the
