--- a/dst/dst_internal.h
+++ b/dst/dst_internal.h
@@ -20,6 +20,7 @@
  */
 #include <limits.h>
 #include <sys/param.h>
+#include <sys/types.h>
 
 #ifndef PATH_MAX
 # ifdef POSIX_PATH_MAX
--- a/includes/arpa/nameser.h
+++ b/includes/arpa/nameser.h
@@ -57,6 +57,8 @@
 #ifndef _ARPA_NAMESER_H_
 #define _ARPA_NAMESER_H_
 
+#include <sys/types.h>
+
 /*
  * Revision information.  This is the release date in YYYYMMDD format.
  * It can change every day so the right thing to do with it is use it
