--- a/Makefile.in
+++ b/Makefile.in
@@ -22,8 +22,8 @@ MV=@MV@
 ENGINE=@SENGINE@
 
 SHELL=/bin/sh
-SUBDIRS=src doc tools
-INSTDIRS=src doc tools
+SUBDIRS=src
+INSTDIRS=src doc
 CLEANSUBDIRS=src tools
 
 CC=@CC@
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -1,20 +1,20 @@
 CC=@CC@
 RM=@RM@
 MV=@MV@
-IRCDLIBS=@LIBS@
+IRCDLIBS=-lresolv -lz @LIBS@ -L$(STAGING_DIR)/lib -L$(STAGING_DIR)/usr/lib
 ENGINE=@SENGINE@
 CRYPTO=@ENCRYPT_SRC@
 INSTALL=@INSTALL@
 INSTALL_BIN=@INSTALL_PROGRAM@
 INSTALL_DIR=@INSTALL_DIR@
 CFLAGS=@CFLAGS@ -fno-strict-aliasing
-CPPFLAGS=@CPPFLAGS@ -I../include
+CPPFLAGS=@CPPFLAGS@ -I../include -I$(STAGING_DIR)/usr/include
 LDFLAGS=@LDFLAGS@
 
-RES_SRC =
+#RES_SRC =
 
 #This is only for very old systems that NEED this
-#RES_SRC = res_mkquery.c res_init.c res_comp.c
+RES_SRC = res_mkquery.c res_init.c res_comp.c
 
 SOURCES = blalloc.c bsd.c channel.c clientlist.c clones.c confparse.c \
           fdlist.c fds.c hash.c hide.c inet_addr.c ircd.c \
@@ -24,7 +24,7 @@ SOURCES = blalloc.c bsd.c channel.c clie
           s_misc.c s_numeric.c s_serv.c s_user.c sbuf.c scache.c send.c \
           struct.c support.c throttle.c userban.c whowas.c zlink.c ssl.c \
 	  bitncmp.c inet_parse_cidr.c m_webirc.c \
-          $(ENGINE) $(CRYPTO)
+          $(RES_SRC) $(ENGINE) $(CRYPTO)
 
 OBJECTS = $(SOURCES:.c=.o) version.o
 
--- a/src/res_comp.c
+++ b/src/res_comp.c
@@ -229,7 +229,7 @@ dn_skipname(u_char *comp_dn, u_char *eom
  * the first name on the list, not the pointer to the start of the
  * message.
  */
-int staticdn_find(u_char *exp_dn, u_char *msg, u_char **dnptrs, 
+static int dn_find(u_char *exp_dn, u_char *msg, u_char **dnptrs,
 		  u_char **lastdnptr)
 {
     u_char *dn, *cp, **cpp;
--- a/tools/Makefile.in
+++ b/tools/Makefile.in
@@ -8,7 +8,7 @@ INSTALL=@INSTALL@
 INSTALL_BIN=@INSTALL_PROGRAM@
 INSTALL_DIR=@INSTALL_DIR@
 CFLAGS=@CFLAGS@ -fno-strict-aliasing
-CPPFLAGS=@CPPFLAGS@ -I../include
+CPPFLAGS=-I../include @CPPFLAGS@
 LDFLAGS=@LDFLAGS@
 
 mkpasswd_SOURCES = mkpasswd.c
