--- a/configure.ac
+++ b/configure.ac
@@ -248,7 +248,7 @@ PKG_CHECK_MODULES([OPENSSL], [openssl])
 # Start by trying to find the needed tinfo parts by pkg-config
 PKG_CHECK_MODULES([TINFO], [tinfo],
   [AC_DEFINE([HAVE_CURSES_H], [1], [Define to 1 if <curses.h> is present])],
-  [PKG_CHECK_MODULES([TINFO], [ncurses],
+  [PKG_CHECK_MODULES([TINFO], [ncursesw],
     [AC_DEFINE([HAVE_CURSES_H], [1], [Define to 1 if <curses.h> is present])],
     [AX_CHECK_LIBRARY([TINFO], [curses.h], [tinfo],
       [AC_DEFINE([HAVE_CURSES_H], [1], [Define to 1 if <curses.h> is present])
