--- a/src/devices/devfreq.h
+++ b/src/devices/devfreq.h
@@ -25,6 +25,7 @@
 #ifndef _INCLUDE_GUARD_DEVFREQ_H
 #define _INCLUDE_GUARD_DEVFREQ_H
 
+#include <sys/time.h>
 #include "device.h"
 #include "../parameters/parameters.h"
 
--- a/src/perf/perf.h
+++ b/src/perf/perf.h
@@ -26,7 +26,7 @@
 #define _INCLUDE_GUARD_PERF_H_
 
 #include <iostream>
-
+#include <cstdio>
 
 extern "C" {
 	#include "../traceevent/event-parse.h"
