--- a/Makefile.am
+++ b/Makefile.am
@@ -2,8 +2,8 @@
 
 ACLOCAL_AMFLAGS=-I m4
 
-SUBDIRS = src examples misc
-DIST_SUBDIRS = $(SUBDIRS) doc wince
+SUBDIRS = src
+DIST_SUBDIRS = $(SUBDIRS) wince
 
 EXTRA_DIST = libdvbpsi.spec libdvbpsi.spec.in libdvbpsi.pc.in bootstrap
 
--- a/Makefile.in
+++ b/Makefile.in
@@ -350,8 +350,8 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = src examples misc
-DIST_SUBDIRS = $(SUBDIRS) doc wince
+SUBDIRS = src
+DIST_SUBDIRS = $(SUBDIRS) wince
 EXTRA_DIST = libdvbpsi.spec libdvbpsi.spec.in libdvbpsi.pc.in bootstrap
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libdvbpsi.pc
