--- a/configure.ac
+++ b/configure.ac
@@ -10,11 +10,14 @@ AC_INIT([yafc],
         [http://yafc-ftp.com])
 AC_CONFIG_SRCDIR(src/main.c)
 AC_CONFIG_LIBOBJ_DIR([lib])
+
+AC_CONFIG_MACRO_DIR([m4])
+AC_CONFIG_AUX_DIR(support)
+
 AC_PROG_CC_C99
+
 AC_USE_SYSTEM_EXTENSIONS
 AC_SYS_LARGEFILE
-AC_CONFIG_MACRO_DIR([m4])
-AC_CONFIG_AUX_DIR(support)
 AM_INIT_AUTOMAKE([subdir-objects])
 AC_CONFIG_HEADERS([config.h])
 
