--- a/ezusb/midi/original/ezusbmidi.c
+++ b/ezusb/midi/original/ezusbmidi.c
@@ -64,7 +64,7 @@ static __xdata __at 0x7e40 byte
 // UART (0) -> EP 1 pipeline
 
 static volatile __data  struct CYCSYN  inserCtl;        // UART->serial
-static __xdata __at 0x7b80              inserBuf[0x40];  // IN7BUF
+static __xdata __at 0x7b80 byte          inserBuf[0x40];  // IN7BUF
 
 static volatile __data  struct CYCSYN  outserCtl;       // serial->UART
 static __xdata __at 0x7b40 byte          outserBuf[0x40]; // OUT7BUF, tmp
@@ -76,7 +76,7 @@ static volatile __data  struct LINSYN  o
 // UART (1) -> EP 1 pipeline
 
 static volatile __data  struct CYCSYN  inserCtl1;        // UART->serial
-static __xdata __at 0x7c00              inserBuf1[0x40];  // IN7BUF
+static __xdata __at 0x7c00 byte          inserBuf1[0x40];  // IN7BUF
 
 // EP 1 OUT -> UART (1) IN pipeline
 
